.github/workflows/ci.yml

```diff
@@ -32,6 +32,29 @@ jobs:
     name: Python Package
     uses: beeware/.github/.github/workflows/python-package-create.yml@main
 
+  docs:
+    name: Check Docs
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3.5.0
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v4.5.0
+      with:
+        python-version: 3.x
+
+    - name: Install tox
+      run: python -m pip install .[dev]
+
+    - name: Check links
+      run: tox -e docs-links
+
+    - name: Check build
+      run: tox -e docs-all
```

Comment:
The RTD task is already doing a docs build; Rather than duplicating the build step, we can add a [`sphinx.fail_on_warning`](https://docs.readthedocs.io/en/stable/config-file/v2.html#sphinx-fail-on-warning) configuration to `.readthedocs.yaml`. 

RTD doesn't expose a "check links" option, or any other way to configure the sphinx arguments AFAICT, so we still need the "docs linting" pass in GitHub Actions.

Reply:
That's true; although, we won't get "nitpick mode" via the `-n` flag in RTD....not exactly the end of the world.

Reply:
All docs-related CI is now happening on RTD.